src/xdplwf/recv.c

```diff
@@ -1586,6 +1588,7 @@ XdpGenericReceivePostInspectNbs(
         //
         if (NET_BUFFER_LIST_FIRST_NB(NblHead) == NbHead) {
             ActionNbl = NblHead;
+            CachedNextNbl = NET_BUFFER_LIST_NEXT_NBL(ActionNbl);
```

Comment:
The essence of this fix is that we're snapshotting `ActionNbl->Next` before it may be appended onto another list, which would invalidate its `Next` pointer, while other NBs still need processing? This makes sense, and kind of falls into the category of the TODO above, aside from the fact the current behavior can lead to bugchecks.

```
        // TODO: Implement NBL splitting such that NBs assigned different
        // actions within an NBL are correctly handled.
```

Reply:
Yup, exactly.